documentation_builder/gapfilling.ipynb

```diff
@@ -0,0 +1,255 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "MAN6PI\n",
+      "GF6PTA\n",
+      "F6PP\n",
+      "TKT2\n",
+      "FBP\n"
+     ]
+    }
+   ],
+   "source": [
+    "solution = gapfill(model, model.objective, universal, demand_reactions=False)\n",
+    "for reaction in solution[0]:\n",
+    "    print(reaction.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "We can obtain multiple possible reaction sets by having the algorithm go through multiple iterations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "---- Run 1 ----\n",
+      "MAN6PI\n",
+      "GF6PTA\n",
+      "F6PP\n",
+      "TKT2\n",
+      "FBP\n",
+      "---- Run 2 ----\n",
+      "MAN6PI\n",
+      "F6PA\n",
+      "GF6PTA\n",
+      "TALA\n",
+      "PGI\n",
+      "---- Run 3 ----\n",
+      "MAN6PI\n",
+      "GF6PTA\n",
+      "F6PP\n",
+      "TKT2\n",
+      "FBP\n",
+      "---- Run 4 ----\n",
+      "MAN6PI\n",
+      "F6PA\n",
+      "GF6PTA\n",
+      "TALA\n",
+      "PGI\n"
+     ]
+    }
+   ],
+   "source": [
+    "result = gapfill(model, model.objective, universal, demand_reactions=False, iterations=4)\n",
+    "for i, entries in enumerate(result):\n",
+    "    print(\"---- Run %d ----\" % (i + 1))\n",
+    "    for e in entries:\n",
+    "        print(e.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "We can also instead of using the original objective, specify a given metabolite that we want the model to be able to produce."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "F6PA\n"
+     ]
+    }
+   ],
+   "source": [
+    "solution = gapfill(model, \"f6p_c\", universal)\n",
+    "for reaction in solution[0]:\n",
+    "    print(reaction.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, note that using mixed-integer linear programming is computationally quite expensive and for larger models you may want to consider using [CORDA](https://github.com/cdiener/corda)."
```

Comment:
Yes, mackinac includes a create_universal_model() function that creates a universal model from a ModelSEED template model that you could use with the gapfill() function. And there is a gapfill_modelseed_model() function that runs the gapfill function on the ModelSEED server.

I should also make a version of cobrababel available as it has functions for creating universal models from other databases.

And it looks like mCADRE has been abandoned too according to the README on github.